cmd/clusterctl/client/init.go

```diff
@@ -199,7 +204,11 @@ func (c *clusterctlClient) setupInstaller(cluster cluster.Client, options InitOp
 
 	providerList, err := cluster.ProviderInventory().List()
 	if err != nil {
-		return nil, err
+		if options.ProviderCRDInstalled {
+			return nil, err
+		} else {
+			return installer, nil
```

Comment:
How about ignoring the error only if the error is becasue the CRD is missing. I think you can check for the errors to be [NoKindMatchError](https://pkg.go.dev/k8s.io/apimachinery@v0.25.0/pkg/api/meta#NoKindMatchError). You should be able to check for that with the `errors.Is()` function.

If the error is becasue the connection to the cluster failed or other network reasons and not necessarily the missing CRD we don't want to ignore it.

Reply:
also, if we ignore the error here we should not return immediately. We should continue with the function execution.